### PR TITLE
Add nodeSelector and tolerations to linkerd-jaeger chart

### PR DESCRIPTION
**Pull request by aatarasoff**
Support `nodeSelector` and tolerations for `linkerd-jaeger` Chart.

There is support to choose nodes for `linkerd` and `linkerd-viz` charts but no support for `linkerd-jaeger` chart.

Properties `nodeSelector` and `tolerations` are added to `values.yml` and processed in templates in the chart.

### CODE REVIEW

**Review by arslanbekov — APPROVED** [review]

**Review by dadjeibaah — COMMENTED** [review]
Thanks for submitting this PR. It looks like CI is failing because the helm docs haven't been updated with the new `nodeSelector` and `tolerations` configuration options. 

You can fix this by running:
``` bin/helm-docs-diff``` and then updating the PR with the generated diff.

**Review by alpeb — COMMENTED** [review]
Thanks @aatarasoff this looks good to me, modulo @dadjeibaah 's observation.

Also, what do you think about providing a bit more flexibility, and let this be configured separately for the injector and the collector? So you'd add `nodeSelector` and `tolerations` under both the `webhook` and `collector` sections in `values.yaml`, and use something like this in the injector 
```
      {{- if .Values.webhook.tolerations -}}
      {{- include "linkerd.tolerations" .Values.webhook | nindent 6 }}
      {{- end -}}
      {{- include "linkerd.node-selector" .Values.webhook | nindent 6 }}
```
(and equivalently in the collector)

**Comment by aatarasoff** [comment]
> Thanks @aatarasoff this looks good to me, modulo @dadjeibaah 's observation.
> 
> Also, what do you think about providing a bit more flexibility, and let this be configured separately for the injector and the collector? So you'd add `nodeSelector` and `tolerations` under both the `webhook` and `collector` sections in `values.yaml`, and use something like this in the injector
> 
> ```
>       {{- if .Values.webhook.tolerations -}}
>       {{- include "linkerd.tolerations" .Values.webhook | nindent 6 }}
>       {{- end -}}
>       {{- include "linkerd.node-selector" .Values.webhook | nindent 6 }}
> ```
> 
> (and equivalently in the collector)

That's a good point. I think that I could add this feature for prometheus (in `linkerd-viz` chart) too.

**Comment by aatarasoff** [comment]
@dadjeibaah I've updated the docs

@alpeb I've made separated values for all components in `linked-jaeger` chart

File: jaeger/charts/linkerd-jaeger/values.yaml [hunk 1]
```diff
@@ -4,21 +4,33 @@ installNamespace: true
...
 
+# -- NodeSelector section, See the
+# [K8S documentation](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector) for more information
+nodeSelector: &default_node_selector
+  beta.kubernetes.io/os: linux
+
+# -- Tolerations section, See the
+# [K8S documentation](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/)
+# for more information
+tolerations: &default_tolerations
```
[the remark this answers is not in the document]
Comment (aatarasoff) [reply jaeger/charts/linkerd-jaeger/values.yaml]: I did not remove comments but I added them to each component entry. As I see now comments are presented on all new values.

**Review by dadjeibaah — APPROVED** [review]
📦 🚢

**Review by alpeb — APPROVED** [review]
Thanks @aatarasoff, looks good to me! :+1:

**Pull request merged**